### ISSUE TITLE
Bluetooth disabled notification after allowing bluetooth acces M1 macOS 12.4

### ISSUE THREAD

**Issue by lesleyblankendal** [emericg/WatchFlower#68]
**Describe the bug**
Bluetooth disabled notification after allowing bluetooth acces macOS 12.4. When the app launches I give it permission but all devices show as offline with a notification that my bluetooth is disabled. Also granted additional access in privacy & settings > bluetooth, to see if that makes any difference.

iPad version that can be run from Mac App Store on M1 iMac doesn't have this issue

**Expected behavior**
Sensors show as online

**To Reproduce**
1. Open app on macOS
2. grant bluetooth access
3. Go to Home Screen

**Screenshots**
<img width="1392" alt="Schermafbeelding 2022-06-29 om 16 18 29" src="https://user-images.githubusercontent.com/4315309/176460759-d4df2a68-9f8b-4554-93ba-c0b3e68b4e64.png">

**Your environment**
Please describe the environment you are using:
1. macOS 12.4, iMac 24 inch late 2020
2. downloaded 4.1 update just now to see if it was fixed. Bug was introduced in 4.0
3. 4.1

**Comment by steveatinfincia** [comment]
I'm seeing this one too, though it works if the binary is launched from a terminal rather than through finder which is certainly strange.

**Comment by jeeftor** [comment]
I also see it on an M1

**Comment by jeeftor** [comment]
>This app has crashed because it attempted to access privacy-sensitive data without a usage description. The app's Info.plist must contain an NSBluetoothAlwaysUsageDescription key with a string value explaining to the user how the app uses this data.

I got this error when I ran it from CLI

**Comment by jeeftor** [comment]
Sounds like a `.plist` issue...

**Comment by jeeftor** [comment]
![image](https://user-images.githubusercontent.com/6491743/207651203-63bb17a1-99c5-4855-911b-41c4c65249c0.png)

But it looks like you have the plist stuff...

**Comment by emericg** [comment]
Indeed. Can you try with a more recent build?
https://github.com/emericg/WatchFlower/actions/runs/3660319742

It has both improvements and newer Qt version, maybe that will help.

**Comment by fabiang** [comment]
Still having this issue with version 5 on Mac OS 13.2.1. Also tried [recent build](https://github.com/emericg/WatchFlower/actions/runs/4637893660) with no luck.

**Comment by emericg** [comment]
Yes I'm really sorry, this problem has proven difficult to understand and to solve...
But I haven't forgotten it. It's just that I failed every time I had a look at it (last time being tonight :)

**Comment by fabiang** [comment]
I've also spent an hour trying to find the error. Imho this must be an upstream bug in Qt and/or OS X?

**Comment by fabiang** [comment]
I think I figured it out. On my local install, I've just added this to my Info.plist:

```
<key>UIBackgroundModes</key>
<array>
   <string>bluetooth-central</string>
</array>
```

Someone able to confirm this? Have it from here: https://stackoverflow.com/a/48390949 Possible value are here: https://developer.apple.com/documentation/bundleresources/information_property_list/uibackgroundmodes

**Comment by emericg** [comment]
That's interesting, but it is said that UIBackgroundModes is only for iOS, and that using it prevents you from being accepted on the app store. Weird option...

I'll test this on a mac but yes we'll need people to confirm this has an actual impact beyond placebo.

**Comment by fabiang** [comment]
@emericg I've just checked if this automatically prevents you from being accepted to the Appstore and that should only be the case if your App doesn't use Bluetooth (or an Apple employee doesn't understand that is does). And from what I've read this values is definitely needed for Bluetooth to function. I've now working the fifth day with this option.

**Comment by emericg** [comment]
It made zero difference in my testing, and the doc still says for iOS only.
But truth be told I don't really care enough to go against this, so I've added this line to the plist, we'll see if anyone report an actual difference in behavior...

**Comment by fabiang** [comment]
Tested once again with [build 7233199927](https://github.com/emericg/WatchFlower/actions/runs/7233199927) and can confirm it's working.

**Comment by NathanBnm** [comment]
I have the same issue with the last build (https://github.com/emericg/WatchFlower/actions/runs/7233199927)
![Capture d’écran 2024-01-03 à 14 34 07](https://github.com/emericg/WatchFlower/assets/45366162/3559a062-a786-493b-aa40-a5ef71bea904)
![Capture d’écran 2024-01-03 à 14 34 28](https://github.com/emericg/WatchFlower/assets/45366162/5867c984-a07b-4f37-98fa-ba816cffc2f2)

**Comment by fabiang** [comment]
I removed the permission manually and now I'm unable to get it working again. Very weird.

**Comment by fabiang** [comment]
Coming back to this after a year...

I've added the following to Info.plist:

```
    <key>NSLocalNetworkUsageDescription</key>
    <string>Test</string>
```

and then adding the permission in the Mac OS settings manually for Bluetooth. Is someone able to confirm that it's working on Mac OS then? I'm currently running OS X 15.1.1 and Watchflower 5.4.

**Comment by NathanBnm** [comment]
@fabiang I can confirm it's working fine after doing this.

I have Watchflower 5.4 on mac M1 with macOS Sequoia 15.3.1

**Comment by fabiang** [comment]
@NathanBnm Could you try to remove the application and try again with the changes in Info.plist? It should not work then again. Still no idea who's fault it is then. I still guess it's how bluetooth is requested by the application.
